report_verification.md

```diff
@@ -0,0 +1,556 @@
+request header `Sec-Private-Aggregation-Private-State-Token`. If this token is
+successfully verified, then the reporting origin has a guarantee that the report
+was associated with a previous request that was signed.
+
+Note: unlike the basic Private State Token API (which enables conveying tokens
+from one site to another), there are no redemption limits for Private
+Aggregation API integration. See [Privacy
+considerations](#privacy-considerations-2) for discussion of other mitigations.
+
+#### Security considerations
+
+This option easily achieves the primary security goals plus some secondary
+security goals. The considerations largely match the Attribution Reporting
+proposal’s given the similar token-based approach, see
+[here](https://github.com/WICG/attribution-reporting-api/blob/main/report_verification.md#security-considerations)
+for details.
+
+#### Privacy considerations
+
+Much like Attribution Reporting’s
+[proposal](https://github.com/WICG/attribution-reporting-api/blob/main/report_verification.md#privacy-considerations),
+this integration is intended to be as privacy-neutral as possible. In
+particular, we want to avoid cross-site information leakage. While each token’s
+issuance occurs using a request from a single site, this token – including its
+metadata, or no token if none was issued – will later be sent with a report from
+a bidder. The identity of which bidders participated in an auction is cross-site
+data.
+
+##### Partitioning can amplify counting attacks
+
+If the count of reports is sensitive, this partitioning could amplify counting
+attacks. However, note that reports can already be partitioned by the
+`scheduled_report_time` and `api` fields. There are designs for [protecting the
+count of encrypted reports](https://github.com/WICG/attribution-reporting-api/blob/main/AGGREGATE.md#hide-the-true-number-of-attribution-reports)
+to mitigate or eliminate the risk of counting attacks. These designs target the
+Attribution Reporting API, but could be adapted for Private Aggregation. Still,
+with less extreme mitigations, there are privacy benefits to reducing the
+partitioning available.
+
+##### Initial design
+
+For the initial design, we do not plan to implement any changes to the Private
+State Token protocol’s public/private metadata bits. So, each token will have
+[six buckets](https://github.com/WICG/trust-token-api/blob/main/ISSUER_PROTOCOL.md#issuance-metadata)
+of metadata embedded. Further, each report could either have a token or no
+token, allowing up to 7 total possibilities (~2.8 bits). This would therefore
+allow the reporting origin to partition its reports into 7 buckets.
+
+##### Potential mitigations
+
+We could consider mitigations in the future. For example:
+
+- restricting the public/private metadata to one bucket – or just a single
+  private bit to avoid an invalid traffic oracle.
+- refusing to send reports to reporting origins using report verification if no
+  token was available/issued.
```

Comment:
```suggestion
  token was available/issued.
- sending null reports with some frequency for buyers that delegate to an issuer who issued tokens.
```

This is pretty speculative but I think we could think of something moderately workable.

Reply:
Yeah, sg to add